tools/modlog/converter.ts

```diff
@@ -530,18 +537,35 @@ export class ModlogConverterTxt {
 		this.database = Database(this.isTesting ? ':memory:' : this.databaseFile);
 		this.database.exec(FS(`databases/schemas/modlog.sql`).readIfExistsSync());
 		if (useFTSExtension || Config.modlogftsextension) {
-			this.database.exec(`SELECT load_extension("native/fts_id_tokenizer.o")`);
-			this.database.exec(`CREATE VIRTUAL TABLE IF NOT EXISTS modlog_fts USING fts5(note, content=modlog, content_rowid=modlog_id, tokenize='id_tokenizer')`);
+			this.database.exec(`SELECT load_extension('native/fts_id_tokenizer.o')`);
+			this.database.exec(`CREATE VIRTUAL TABLE IF NOT EXISTS modlog_fts USING fts5(note, userid, autoconfirmed_userid content=modlog, content_rowid=modlog_id, tokenize='id_tokenizer')`);
+			this.database.exec(`CREATE VIRTUAL TABLE IF NOT EXISTS alts_fts USING fts5(userid, content=alts, tokenize='id_tokenizer')`);
 		} else {
-			this.database.exec(`CREATE VIRTUAL TABLE IF NOT EXISTS modlog_fts USING fts5(note, content=modlog, content_rowid=modlog_id)`);
+			this.database.exec(`CREATE VIRTUAL TABLE IF NOT EXISTS modlog_fts USING fts5(note, userid, autoconfirmed_userid, action_taker_userid, content=modlog, content_rowid=modlog_id)`);
+			this.database.exec(`CREATE VIRTUAL TABLE IF NOT EXISTS alts_fts USING fts5(userid, content=alts)`);
 		}
 
 		this.insertionQuery = this.database.prepare(
 			`INSERT INTO modlog (timestamp, roomid, visual_roomid, action, userid, autoconfirmed_userid, ip, action_taker_userid, note)` +
 			` VALUES ($time, $roomID, $visualRoomID, $action, $userid, $autoconfirmedID, $ip, $loggedBy, $note)`
 		);
-		this.FTSInsertionQuery = this.database.prepare(`INSERT INTO modlog_fts (rowid, note) VALUES (?, ?)`);
+		this.noteInsertionQuery = this.database.prepare(`INSERT INTO modlog_fts (rowid, note) VALUES (?, ?)`);
+		this.useridInsertionQuery = this.database.prepare(`INSERT INTO modlog_fts (rowid, userid) VALUES (?, ?)`);
+		this.autoconfirmedInsertionQuery = this.database.prepare(
+			`INSERT INTO modlog_fts (rowid, autoconfirmed_userid) VALUES (?, ?)`
+		);
+		this.actionTakerInsertionQuery = this.database.prepare(
+			`INSERT INTO modlog_fts (rowid, action_taker_userid) VALUES (?, ?)`
+		);
```

Comment:
Surely if you ran these in succession, you'd get a constraint error? Is there something I'm missing here? You've tested that this works, right?

Reply:
No, the converter works fine without constraint errors on my end.

Reply:
I see... this seems to be undocumented behavior with contentless FTS tables
```
sqlite> create virtual table foo using fts5(bar, baz);
sqlite> insert into foo (rowid, bar) values (1, 'test');
sqlite> insert into foo (rowid, baz) values (1, 'test');
Error: constraint failed
sqlite> create table foo2 (bar, baz);
sqlite> create virtual table foo_fts using fts5(bar, baz, content=foo2);
sqlite> insert into foo2 values ('test', 'test');
sqlite> insert into foo_fts (rowid, bar) values (1, 'test');
sqlite> insert into foo_fts (rowid, baz) values (1, 'test');
sqlite> select * from foo_fts('test');
test|test
```
I'd be careful about relying on this sort of thing, the FTS module seems to accept whatever you throw at it, even if it causes problems down the line:
```
sqlite> insert into foo_fts(rowid, bar) values (100, 'crash please');
sqlite> select * from foo_fts;
test|test
sqlite> select * from foo_fts('crash');
Error: database disk image is malformed
```
Is there an issue with inserting multiple columns in the same statement? Wouldn't it simplify the code a bit too?

Reply:
sure, that's fine. That's a bit worrying regarding illegal insertions being allowed through FTS though.